osdep/als.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Generic interface for ambient light sensors (ALS)
+ *
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "osdep/als.h"
+#include "player/core.h"
+
+#ifndef __linux__
+#include "ta/ta_talloc.h"
+#include "video/out/vo.h"
+#endif
+
+#ifndef __linux__
+struct mp_voctrl_als {
+    MPContext *mpctx;
+};
+#endif
+
+mp_als *mp_als_create(void *parent, MPContext *mpctx) {
+#ifdef __linux__
```

Comment:
Could you follow the existing practice in osdep? There should not be ifdefs and instead meson should select file with the implementation suitable for given platform.